packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -259,22 +326,55 @@ class FlatTermSelector extends Component {
 			_x( 'Remove %s', 'term' ),
 			singularName
 		);
+		const tagsButtonDescription = _x(
+			'Choose from the most used tags.',
+			'term'
+		);
 
 		return (
-			<FormTokenField
-				value={ selectedTerms }
-				suggestions={ termNames }
-				onChange={ this.onChange }
-				onInputChange={ this.searchTerms }
-				maxSuggestions={ MAX_TERMS_SUGGESTIONS }
-				disabled={ loading }
-				label={ newTermLabel }
-				messages={ {
-					added: termAddedLabel,
-					removed: termRemovedLabel,
-					remove: removeTermLabel,
-				} }
-			/>
+			<div>
+				<FormTokenField
+					value={ selectedTerms }
+					suggestions={ termNames }
+					onChange={ this.onChange }
+					onInputChange={ this.searchTerms }
+					maxSuggestions={ MAX_TERMS_SUGGESTIONS }
+					disabled={ loading }
+					label={ newTermLabel }
+					messages={ {
+						added: termAddedLabel,
+						removed: termRemovedLabel,
+						remove: removeTermLabel,
+					} }
+				/>
+				<div className="editor-post-taxonomies__flat-terms-related-list-wrapper">
+					<Button
+						className="is-link"
+						onClick={ () => this.toggleRelatedTerms() }
+						aria-expanded={ ! this.state.areRelatedTermsHidden }
+					>
+						{ tagsButtonDescription }
+					</Button>
+					{ this.state.areRelatedTermsHidden === false ? (
+						<ul className="editor-post-taxonomies__flat-terms-related-list">
```

Comment:
Please add a `role="list"` attribute. Safari and VoiceOver don't announce styled lists as lists. That's why Classic Editor explicitly sets this attribute on the list. It is technically redundant but it's the only way to make Safari and VoiceOver announce the number of items within the list and their position e.g. "n of n".

Classic editor: 

<img width="954" alt="Screenshot 2020-05-02 at 11 12 56" src="https://user-images.githubusercontent.com/1682452/80860176-5a6f0700-8c66-11ea-9db1-4636e7fa32d9.png">

Reply:
I agree with all of your Screen Reader improvements, except for the Term added one, since this is part of the FormTokenField element and thus is not related to this PR.

Reply:
Not sure that's correct. In the Classic Editor, it is true that the tags "cloud" does use the same `speak` message used for the "Add tag" input field. However, clicks on tags in the tag cloud trigger a message that belongs to the tag cloud. Pretty sure it works this way, since I coded it :) See https://core.trac.wordpress.org/changeset/38880
Thus, the message does belong to the tags cloud. Not implementing it would be a lack of parity with the Classic Editor and an accessibility regression.